presto-hive/src/main/java/com/facebook/presto/hive/HiveNodePartitioningProvider.java

```diff
@@ -68,6 +68,10 @@ public ConnectorBucketNodeMap getBucketNodeMap(ConnectorTransactionHandle transa
         HivePartitioningHandle handle = (HivePartitioningHandle) partitioningHandle;
         NodeSelectionStrategy nodeSelectionStrategy = getNodeSelectionStrategy(session);
         int bucketCount = handle.getBucketCount();
+        if (sortedNodes.size() == 0) {
```

Comment:
would it make sense to move this `if` to line 78? e.g. , something like 
```
case SOFT_AFFINITY:
    if (sortedNodes.size() != 0) {
        return createBucketNodeMap(Stream.generate(() -> sortedNodes).flatMap(List::stream).limit(bucketCount).collect(toImmutableList()), nodeSelectionStrategy);
     }
     // Otherwise, create bucket node map as if there is no preference  
```

Also curious when will `sortedNodes` be empty ? -- that means there is no worker?